pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/ColumnMetadata.java

```diff
@@ -90,7 +90,23 @@ default boolean isMinMaxValueInvalid() {
   @Nullable
   Set<Integer> getPartitions();
 
-  Map<IndexType<?, ?, ?>, Long> getIndexSizeMap();
+  long getIndexSizeFor(IndexType type);
+
+  default void addIndexSize(short indexType, long size) {
```

Comment:
The methods in this interface should be read-only

Reply:
How do I then add indexes to it ?
In master branch, indexes are added via :  
```java
_columnMetadataMap.get(parsedKeys[0]).getIndexSizeMap().put(indexType, mapConfig.getLong(key));
```

Reply:
This is an interface, and all the APIs are read only.
Consider adding it only to `ColumnMetadataImpl.Builder`

Reply:
I moved method to ColumnMetadataImpl because Builder doesn't fit code in SegmentMetadataImpl.